### DISCUSSION TITLE
performance.now() doesn't work in snap server

### DISCUSSION THREAD

**Discussion by porco-rosso-j** [MetaMask/snaps#2476] [Q&A] [answered]
One of the dependencies in the snap I develop ([`packages/snap`](https://github.com/porco-rosso-j/aztec-snap/tree/dev/packages/snap)) uses `performance.now()` instead of `Date.now()` to obtain the current time. But this causes an error in snap server app. 

<img width="590" alt="Screenshot 2024-06-08 at 20 09 39" src="https://github.com/MetaMask/snaps/assets/88586592/5916f10e-f8dd-46c2-82ef-849d09840922">

I'm wondering if there is walkaround to resolve this besides patching the package i use. Adding `performance-polyfill` in the snap config didn't help. 

```
  customizeWebpackConfig: (config) =>
    merge(config, {
      resolve: {
        fallback: {
          performance: require.resolve('performance-polyfill'),
        },
      },
    }),
```

**Comment by FlaviuWeb3** [comment, marked as answer]
Hi @porco-rosso-j Performance.now is not available in Snaps. Also please note that they both serve different purposes.

performance.now() is relative to page load and more precise in orders of magnitude. Use cases include benchmarking and other cases where a high-resolution time is required such as media (gaming, audio, video, etc.)

It should be noted that performance.now() is only available in newer browsers (including IE10+).
See -> https://stackoverflow.com/questions/30795525/performance-now-vs-date-now
